### ISSUE TITLE
Bind legacy two-segment ORT segment execution to manifest-verified artifacts

### ISSUE THREAD

**Issue by azumag** [azumag/unzen#1194]
## Problem

`LLM-proto/tools/verify_split_onnx.py::verify_split()` now pins the **full-model reference** through `verified_source_execution_snapshot()` (#1182), but the two generated split sessions are still opened directly from caller-provided pathnames:

```python
segment0_session = ort.InferenceSession(str(segment0_path), providers=providers)
...
segment1_session = ort.InferenceSession(str(segment1_path), providers=providers)
```

The legacy `unzen-real-two-segment-onnx` manifest already records `segments[0/1].path` and `sha256`, and source/external-data provenance. However `verify_split()` currently only takes a stable manifest snapshot, reads boundary/logits metadata from it, and does not verify that the CLI `--segment0/--segment1` pathnames are the manifest-declared graph generation before ORT opens them. A same-byte/different-inode replacement or an in-place mutation after any ad-hoc check could therefore make the legacy numerical result describe a different segment generation than the manifest.

#1193 closes this exact verified-generation -> ORT-open boundary for the newer multi-segment logits and KV-decode verifiers with `artifact_execution_snapshot.py`. The legacy gate should adopt the same guarantee without changing its two-segment public CLI/report contract.

## Acceptance criteria

- Validate that the legacy manifest kind/schema is appropriate and bind the executed segment graph identities to the manifest-declared `segments[0/1].path` + SHA-256 rather than trusting independent pathnames.
- Either require the supplied `--segment0/--segment1` paths to resolve to the manifest-declared files or deprecate their authority in favor of manifest-relative paths; preserve CLI compatibility unless there is a documented reason not to.
- Execute both split `InferenceSession`s from a pinned same-filesystem snapshot/hard-link tree (reuse/refactor the #1193 helper where sensible), preserving external-data relative resolution without copying multi-GiB payloads.
- Fail closed on same-byte/different-inode pathname replacement after verification and on in-place graph/external-data mutation while ORT can observe the artifact.
- Keep the existing full-model source snapshot, provider selection, tolerances, boundary routing, report schema, and sequential full-model -> segment0 -> segment1 session lifetime unchanged.
- Cleanup must remain identity-bound: if the temporary execution workspace pathname is renamed/replaced, do not recursively delete the replacement tree.
- Add focused regressions for path/manifest mismatch, same-byte segment replacement, in-place mutation, snapshot-backed ORT paths, and normal cleanup.
- Update the legacy two-segment numerical-verification docs to distinguish source-model provenance from split-artifact provenance.

Host-side numerical-evidence integrity only. No production deployment, credentials, billing, #908 publication-concurrency decision, or #877 legacy deadline-policy change.

Related: #1192/#1193, #1182, #167.
